app/_src/gateway/how-kong-works/load-balancing.md

```diff
@@ -203,49 +259,75 @@ Supported hashing attributes are:
+- the cookie based approach will work well for browser based requests, but less so
+  for machine-2-machine clients which will often omit the cookie.
+- avoid using hostnames in the balancer as the
+  balancers might/will slowly diverge because the DNS ttl has only second precision
+  and renewal is determined by when a name is actually requested. On top of this is
+  the issue with some nameservers not returning all entries, which exacerbates
+  this problem. So when using the hashing approach in a Kong cluster, preferably add
+  `target` entities by their IP address. This problem can be mitigated by balancer
+  rebuilds and higher ttl settings. 
 
-{% endif_version %}
 
-The ring-balancer also supports the `least-connections` algorithm, which selects
-the target with the lowest number of connections, weighted by the Target's
-`weight` attribute.
+### Least-Connections
+
+This algorithm keeps track of the number of in-flight requests for each backend.
+The weights are used to calculate "connection-capacity" of a backend. Requests are
+routed towards the backend with the highest spare capacity.
+
+#### Considerations
+
+- good distribution of traffic.
+- does not improve cache-hit ratio's.
+- more dynamic since slower backends will have more connections open, and hence
+  new requests will be routed to other backends automatically.
+
+{% if_version gte:3.2.x %}
+### Latency
+
+The `latency` algorithm is based on peak EWMA (exponentially weighted moving average),
+which ensures that the balancer selects the backend by lowest latency
+(`upstream_response_time`). The latency metric used is the full request cycle, from
+TCP connect to body response time. Since it is a moving average, the metrics will
+"decay" over time.
+
+Weights will not be taken into account.
+
+#### Considerations
+
+- good distribution of traffic provided there is enough base-load to keep the
+  metrics alive (since they are "decaying")
+- not suitable for long-lived connections like websockets or server-sent events (SSE)
+- very dynamic since it will constantly optimize.
+- ideally this works best with low variance in latencies. This means mostly similar
+  shaped traffic and even workloads for the backends. For example usage
+  with a GraphQL backend serving small-fast queries as well big-slow ones will result
+  in high variance in the latency metrics, which will skew the metrics.
+- properly set up the backend capacity and ensure proper network latency to prevent
+  resource starvation. For example; 2 servers, one a small capacity close by (low
+  network latency), the other high capacity far away (high latency). Most traffic
+  will be routed to the small one, until its latency starts going up. The latency
+  going up however means the small server is most likely suffering from resource
+  starvation. So in this case the algorithm will keep the small server in a constant
+  state of resource starvation, which is most likely not efficient.
```

Comment:
```suggestion
- properly set up the backend capacity and ensure proper network latency to prevent
  resource starvation. For example, use 2 servers: one a small capacity close by (low
  network latency), the other high capacity far away (high latency). Most traffic
  will be routed to the small one, until its latency starts going up. The latency
  going up however means the small server is most likely suffering from resource
  starvation. So, in this case, the algorithm will keep the small server in a constant
  state of resource starvation, which is most likely not efficient.
```